### PR TITLE
Reload data sets in Muon Analysis without crash

### LINKED ISSUE #27881: Instrument View scrolling between files
**Original reporter:** James Lord
### Expected behavior
Data sets should be reloaded cleanly to allow the pointer to Instrument View to survive.
### Actual behavior
Reloading files with the Muon Analysis GUI in Workbench can crash Mantid when Instrument View is open for theses files!
### Steps to reproduce the behavior
Open Workbench
Open Muon Analysis GUI
Type in 62260 as the run number
Show Instrument on the "EMU62260_raw_data MA" workspace
Enter the Pick tab and click on some detectors
go back to the Muon Analysis GUI and use the <> buttons to move to the next run, and then back to 62260. Crash.
### Platforms affected
Workbench all

### PR DESCRIPTION
**Pull request by giovannidisiena**
**Description of work.**
This PR allows data sets to be reloaded within the Muon Analysis GUI (Workbench) without crashing Mantid while the Instrument View is open.

**To test:**
- Open Workbench
- Open Muon Analysis GUI
- Type in 62260 as the run number
- Show Instrument on the "EMU62260_raw_data MA" workspace
- Enter the Pick tab and click on some detectors
- Go back to the Muon Analysis GUI and use the <> buttons to move to the next run, and then back to 62260
- Should **not** crash

Fixes #27881 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by cailafinn — CHANGES_REQUESTED** [review]
Having some trouble with functional testing due to increment button not working on Ubuntu. I do have a code comment though.

File: qt/widgets/instrumentview/inc/MantidQtWidgets/InstrumentView/InstrumentWidget.h [hunk 1]
```diff
@@ -312,6 +314,9 @@ private slots:
   /// dict of selected tabs
   std::vector<std::pair<std::string, bool>> m_stateOfTabs;
 
+private:
+  bool m_wsReplace;
+
 private:
   /// ADS notification handlers
```
[the remark this answers is not in the document]
Comment (giovannidisiena) [reply qt/widgets/instrumentview/inc/MantidQtWidgets/InstrumentView/InstrumentWidget.h]: Yes, it is to keep the member variable separate from the functions below.

**Comment by giovannidisiena** [comment]
@ConorMFinn I am going to look into the increment button issue. As we discovered yesterday, it wasn't a problem for me initially (maybe an old master) but now it is.

File: qt/widgets/instrumentview/inc/MantidQtWidgets/InstrumentView/InstrumentWidget.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (ConorMFinn) [reply qt/widgets/instrumentview/inc/MantidQtWidgets/InstrumentView/InstrumentWidget.h]: There is another member variable at Line 346. These should be together. Whether you move your new one down or the existing one up is down to you, but it should match the surrounding code.

**Review by cailafinn — APPROVED** [review]
This looks good to me now. Crash no longer occurs.

**Pull request merged**